### PR TITLE
vtk: fix build on < 10.7

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Fix

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by Dave-Allured — APPROVED** [review]
Nice work!  I leave this in your hands to continue debugging if problems arise on the builders.  I will help when I can.

File: graphics/vtk/Portfile
Comment (Dave-Allured) [inline graphics/vtk/Portfile:1]: For my education only, why did you move up `compiler_blacklist_versions`?

Comment (barracuda156) [reply graphics/vtk/Portfile:1]: Alphabetically, but if you dislike that, I will drop it, of course.

Comment (Dave-Allured) [reply graphics/vtk/Portfile:1]: No, leave it the way you did it.  I just thought there might be some ordering dependency that I needed to know about.  This is good.

Comment (barracuda156) [reply graphics/vtk/Portfile:1]: Ok, thanks

**Review by reneeotten — CHANGES_REQUESTED** [review]
please add links to the upstream fixes / PRs / issues - there is no easy way for anyone reviewing this to verify whether upstream agrees with any of these suggested changes.

**Comment by barracuda156** [comment]
> please add links to the upstream fixes / PRs / issues - there is no easy way for anyone reviewing this to verify whether upstream agrees with any of these suggested changes.

Upstream has deliberately broken it several years ago. And that was not an issue for anyone here, until we try fixing it :) Every time I am slightly surprised how breaking something is perfectly fine and often gets committed straight to the master without even a nominal PR, but fixing requires an external approval.
The worst thing which can happen here is if Mesa’s GL claims to be compatible but is not really for some cases, so possibly OpenGL support will still need some extra work. Don’t know why we need a sanction from upstream to fix their bugs locally for us.

But anyway, there is a repo on Gitlab, I can make a PR for portable fixes and an issue for non-portable (I am not gonna try to figure out how to add a conditional fix for atomics in this proprietary syntax myself).

**Comment by barracuda156** [comment]
@reneeotten I have added refs.

**Pull request merged**